### PR TITLE
Network 2018 04 01

### PR DESCRIPTION
**Pull request by MikhailTryakhov**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
+@dsgouda

File: specification/network/resource-manager/Microsoft.Network/stable/2018-02-01/ddosProtectionPlan.json [hunk 1]
```diff
@@ -258,7 +258,35 @@
   },
   "definitions": {
     "DdosProtectionPlan": {
+      "description": "A DDoS protection plan in a resource group.",
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-02-01/ddosProtectionPlan.json]: @MikhailTryakhov please undo any changes to the 2018-02-01 directory files.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/applicationSecurityGroup.json [hunk 2]
```diff
@@ -0,0 +1,319 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2018-02-01"
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/applicationSecurityGroup.json]: Invalid ApiVersion for this folder. Same pattern applies to all files that are currently 2018-02-01 if necessary (don't touch VMSS files)

**Review by lmazuel — CHANGES_REQUESTED** [review]

**Comment by MikhailTryakhov** [comment]
@dsgouda  @lmazuel fixed requested changes. please review

**Comment by dsgouda** [comment]
@lmazuel nice catch!
Any other comments I might have missed?

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@MikhailTryakhov once @lmazuel approves we're good to merge

**Comment by MikhailTryakhov** [comment]
Thanks @dsgouda. Can we create May branch right after that?
@lmazuel to make review easier. I've changed a version from 2018-02-01 to 2018-04-01 in all files of network branch except VMSS (2 files). Please let me know if there something else to change

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/ApplicationGatewayAvailableSslOptionsGet.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "parameters": {
+    "api-version": "2018-02-01",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/ApplicationGatewayAvailableSslOptionsGet.json]: Please update ApiVersion of examples as well

**Review by lmazuel — APPROVED** [review]
I approve to unlock, since last thing is a detail, but api-version of example needs to be in sync of the Swagger content.

**Pull request merged**